### ISSUE TITLE
Additional no-connects on GC-6052S sample

### ISSUE THREAD

**Issue by Bob131** [mikilopez/clockdisplay#2]
It would appear that only the right-most segments of the most significant digit are functional. I'm not sure whether this module is just defective (it was pulled out of a cheap/rubbishy ~10 year old alarm clock) or maybe it's a different PCB revision. Attached are some pictures, in case you can identify the module as being (dis)similar to the one you documented.

Sorry for the shaky cam and general poor image quality. The number on the back is GC-6052L-S, in case it's a bit too hard to read.

![gc-6502s_front](https://user-images.githubusercontent.com/4337852/31703433-e0938238-b41f-11e7-9098-e005893c3271.jpg)
![gc-6502s_back](https://user-images.githubusercontent.com/4337852/31703442-e7fd371c-b41f-11e7-8e13-1225763f9296.jpg)

**Comment by mikilopez** [comment]
Hello.
This is my fault, I guess, because I didn't explain properly how to connect the Arduino to the display (is a missing thing in the PDF file).

Let's say we have this constructor from the examples:
`clockdisplay cd(4, 2, 3, GC_6052S);`

This means you connect Arduino pins 2 and 3 to display pins 1 and 2 (each one through a transistor), and Arduino pins 4 to 17 (fourteen pins total) to the 14 segment pins of the display. For the GC_6052S display, pins are 6, 7, 8, 9, 10, 12, 13, 15, 16, 17, 18, 19, 20, 21 (fourteen pins total).

All of this is "as far as I remember", I did this thing many years ago and I don't have the material right now for a quick test. A shame I didn't take pics of the display, but your one is good enough.

Hope this helped you!
Regards.

**Comment by Bob131** [comment]
Oh, my bad. I just noticed I neglected to mention what the issue was actually about. Let me try again.

In the models.pdf, the most significant digit on the GC-6052S diagram has one segment labeled 'nc', with segments connected to pins 7 and 8. In contrast, my module has pins 7 and 8 also unconnected. The issue was specifically about this discrepancy, but I guess there isn't much we can do about it if you don't have the module handy.

(As an aside, looking at the alarm clock I pulled this out of it's clear at some point in the product's life these pins were used: they're commoned together and brought out to the land pattern of the clock controller, but that pin of the controller was simply bent out of shape before installation to prevent a connection)

Sorry for the confusion. I swear it made sense when I wrote it ;)

**Comment by mikilopez** [comment]
Uhm, interesting.
You are in the US, right?
Is it (un)common to find alarm clocks able to display "20:00" when it's 8:00pm?
Because I think your display can only show a "1" on the first digit.

Here in Spain we usually use the 24h format (from 0:00h to 23:59h), so my display can show a "1" and a "2" in the first digit. And that's why the top-left segment is a no-connect: it's never used.

If you take a look at the other documented model, you can show the digit "2" using only pins 15 and 16 (the 16 rules it).

That's the way cheap stuff works.

**Concluding**: I'm sure my pins 7 and 8 worked, so one of these cases apply to your display:
* Pins 7 and 8 are broken (unlikely)
* You have a different PCB revision, adapted to you country (likely)

**Comment by Bob131** [comment]
>You are in the US, right?

Australia.

>Is it (un)common to find alarm clocks able to display "20:00" when it's 8:00pm?

Humm, I'm not sure. I think most people prefer 12 hour time, but 24 hour time is pretty universal on consumer goods and most government services use 24 hour time nowadays. AFAIR this clock didn't do 24 hour time, but I have another that does, so ¯\\\_(ツ)_/¯

>Because I think your display can only show a "1" on the first digit.

Indeed.

Would you mind if this issue was left open? Just in case some corroborating information crops up one day.

Cheers!

**Comment by mikilopez** [comment]
Sure! Ever-open issue from now on.
Thanks for the contribution!

**Comment by blueduckjf** [comment]
Hello from the future! I recently got a GC-6052L-S display out of a cheap alarm clock from Goodwill (I am in the US). I can confirm Bob's description of pins 7 and 8 not being connected to anything. I agree that it is probably because 24-hour time is not commonly used in the US, so only the rightmost segments of the most significant digit light.  All other documented pins seem correct.

Additionally, to get the center colon to light, pin 30 is the anode and pin 1 is the cathode.  To get the top leftmost dot to light, pin 5 in the anode and pin 1 is the cathode.  I haven't figured out how to get the bottom leftmost dot or the rightmost dot to light, but I haven't looked into it closely.

It's worth nothing in my panel, the pins for the digits are connected with a ribbon cable and pin 30 is a separate wire.  Pics attached.
![image0](https://user-images.githubusercontent.com/23023219/113516572-f7a67f80-9548-11eb-9aed-71d39d73ba8b.jpeg)
![image2](https://user-images.githubusercontent.com/23023219/113516573-f8d7ac80-9548-11eb-9220-620c84b8e64a.jpeg)

**Comment by mikilopez** [comment]
Hi, traveller from the future!
Is it a coincidence that I just found my display in a drawer one month ago and started to play with it?
I found out that the code doesn't work anymore, so I want to rewrite everything from scratch, and remove the latex files and use the Github wiki to document the models. Maybe include some pictures (can I use your pictures?), add some special functions (like "standar7segmentTodisplay").

Also, I realized that your display or Bob's have a "2" or "Ver2". Mine doesn't. So that's problably the difference I'll document that.

I also had the ribbon and the pin 30 cable, though I removed the ribbon and soldered some straight pins (easier to use on bredboard).

Thanks for your contribution! :smiley: 

EDIT: I also noticed that your display and Bob's back label says "GC-6052L-S", while mine says "GC-6052X-S"

![IMG_20210405_130535](https://user-images.githubusercontent.com/3891368/113568387-c1343780-9610-11eb-9f7a-c6c798140d39.jpg)
![IMG_20210405_130517](https://user-images.githubusercontent.com/3891368/113568966-e83f3900-9611-11eb-9a4f-6a83023f193e.jpg)

**Comment by mikilopez** [comment]
The additional no-connects are there because the board is a slightly different version, labeled as "GC-6052S 2" or "GC-6052S ver 2". This version can only display time in 12h format (common mainly in US), not 24h (common mainly in european countries).
I'll update the project to include this version.
